src/ThemeCatalog.js

```diff
@@ -0,0 +1,843 @@
+import {imagePath} from './constants';
+
+import './styles/ThemeCatalog.scss';
+
+const ThemeCatalog = () => {
+  return (
+    <ComponentCatalogPanel
+      hero={<ThemeHero />}
+      title='Theme'
+      description='Color in Material Design is inspired by bold hues juxtaposed with muted environments, deep shadows, and bright highlights.'
+      designLink='https://material.io/go/design-color-theming'
+      docsLink='https://material.io/components/web/catalog/theme/'
+      sourceLink='https://github.com/material-components/material-components-web/tree/master/packages/mdc-theme'
+      demos={<ThemeDemos />}
+    />
+  );
+}
+
+class ThemeHero extends Component {
+  componentInstances = [];
+  themeStylesheetFilename = null;
+  initRipple = ele => ele && this.componentInstances.push(new MDCRipple(ele));
+
+  componentWillMount() {
+    if(!this.themeStylesheetFilename) {
+      // Retrieve the location of the compiled theme.css file
+      fetch('asset-manifest.json')
+          .then(result => result.json())
+          .then(data => this.themeStylesheetFilename = data['theme.css'])
+          .then(() => this.appendThemeStyle());
+    } else {
+      this.appendThemeStyle();
+    }
+  }
+
+  /**
+   * Adds the theme stylesheet to the document head to override the
+   * catalog theme.
+   */
+  appendThemeStyle() {
+    if(this.themeStylesheetFilename) {
+      const styles = document.createElement('link');
+      styles.setAttribute('rel', 'stylesheet');
+      styles.setAttribute('type', 'text/css');
+      styles.setAttribute('id', 'themeStylesheet');
+      styles.setAttribute('href', this.themeStylesheetFilename);
+      document.getElementsByTagName('head')[0].appendChild(styles);
+    }
+  }
+
+  componentWillUnmount() {
+    this.componentInstances.forEach(ripple => ripple.destroy());
+
+    // Remove the theme stylesheet if present
+    document.head.removeChild(document.getElementById('themeStylesheet'));
+  }
+
+  render() {
+    return (
+      <div className='theme-demo'>
+        <button className='hero-button mdc-button' ref={this.initRipple}>
+          Text
+        </button>
+        <button className='hero-button mdc-button mdc-button--raised' ref={this.initRipple}>
+          Raised
+        </button>
+        <button className='hero-button mdc-button mdc-button--outlined' ref={this.initRipple}>
+          Outlined
+        </button>
+      </div>
+    );
+  }
+}
+
+class ThemeDemos extends Component {
+  componentInstances = [];
+
+  // Component specific initialization functions
+  initCheckbox = ele => ele && this.componentInstances.push(
+      new MDCCheckbox(ele));
+  initIconToggle = ele => ele && this.componentInstances.push(
+      new MDCIconToggle(ele));
+  initLinearProgress = ele => {
+    if (!ele) return;
+    const linearProgress = new MDCLinearProgress(ele);
+    this.componentInstances.push(linearProgress);
+    linearProgress.progress = 0.5;
+    if (ele.dataset.buffer) {
```

Comment:
Nit: check out linear progress catalog page for a better way of initializing buffered indicators